doc/design/vic-machine/configure.md

```diff
@@ -0,0 +1,182 @@
+
+## Update In Progress Status
+Same to vic-machine upgrade, concurrent configure requests for same VCH is not allowed. vic-machine should be able to detect if one VCH is running configure/upgrade. If true, return useful error message. And also should not leave that configure flag in there after configure is stopped, or even interrupted.
+
+Issue https://github.com/vmware/vic/issues/4069 is tracking this problem.
+
+## Backward Compatibility
+It's possible that after vic engine released new version, user is trying to configure older version's VCH with newer vic-machine.
+
+There are few options:
+- Run configure using same version's vic-machine (this option should technically always work)
+
+  For any customer that does not want to change vic-machine version, they need to run configure using same version's vic-machine.
+
+- Combine upgrade process into configure
+
+  Upgrade is actually one special kind of configure, technically it's easy to have upgrade run together with configure. User might want to configure VCH, and also want to upgrade it after or before configure. In that case, a combined configure and upgrade command can provide a better user experience.
+
+ Here is the combined command ```vic-machine configure --upgrade <all configure and upgrade options supported here>```
+
+- Support backward compatibility in vic-machine
+
+  This will need to add backward compatible code in vic-machine. Even initially there is only few changes, that will expand massively over time, which is not consistent with our data migration design, so this is not our option.
+
+# VCH configure Rollback (P0.5)
+Similar to VCH upgrade workflow, before configure VCH, snapshot should be took to keep existing status. If the configure failed eventually, VCH should be reverted to old status automatically.
+
+The Configure Rollback is talking about after a successful configure, vic-machine is still able to rollback VCH's old configuration. The command is like vic-machine configure --rollback
+
+This feature is nice to have, but as vic-machine upgrade already have this capability, mark it as P0.5.
+
+# Manifest File
+vic-machine create already has more than 50 options, and the number will increase over time. With this configure feature added, most of those options need to be updated as well. To improve user experience, in my opinion, we need Manifest file to simplify vic-machine command generation.
+
+[vic-machine design](vic-machine.md#vch-manifest) mentioned manifest idea, here we'll start from a simple implementation. The manifest file format will be same to the yml file format, mentioned in [configuration inspection](#configuration-inspection)
+
+- vic-machine inspect --manifest filename will dump all configurations in yml file format to the manifest file.
+- vic-machine create --manifest will load the manifest file and create VCH based on it.
+- vic-machine configure --manifest will load the manifest file and update VCH configuration based on it.
```

Comment:
I would also like us to be able to use a manifest as a base, with CLI options to refine or specialize. For example the following would use the same semantics as `configure` to merge the CLI options with the configuration specified by the manifest.:
```
vic-machine create --manifest=conf.yml --name=new-vch --memory=12GB
```

This allows both easy use of manifests as templates and positions us well for adding authority delegation mechanisms to the manifests.